Zend/zend_compile.c

```diff
@@ -8067,6 +8067,27 @@ static zend_string *zend_begin_method_decl(zend_op_array *op_array, zend_string
 		zend_error(E_COMPILE_WARNING, "Private methods cannot be final as they are never overridden by other classes");
 	}
 
+	if ((fn_flags & ZEND_ACC_ABSTRACT) &&
+		!(ce->ce_flags & (ZEND_ACC_EXPLICIT_ABSTRACT_CLASS|ZEND_ACC_TRAIT)) &&
+		!in_interface
+	) {
+		// Don't say that the class should be declared abstract if it is
+		// anonymous and can't be abstract
+		const char *msg;
+		if (ce->ce_flags & ZEND_ACC_ANON_CLASS) {
+			msg = "Anonymous class %s cannot contain abstract method %s::%s";
```

Comment:
I'm not sure there's a point in printing the class name twice? I don't have a strong opinion on this. Maybe @kocsismate does?

Reply:
I'm not sure there is a point other, but I couldn't really figure out a better way to do it - the message is about the class not being allowed to have abstract methods, and then the method name is usually given with the class name too. What would you think of `...abstract method ::foo` without the class name?

Reply:
E.g. "Must not declare abstract method Foo::bar() in anonymous class" would be an option. But it's not a big deal.